src/utils/fileUpload.js

```diff
@@ -45,27 +45,19 @@ const findUniquePath = async function(client, userRoot, path) {
 	if (await client.exists(userRoot + path) === false) {
 		return path
 	}
+
 	// Get the file extension (if any)
 	const fileExtension = getFileExtension(path)
-	// If there's a file extention, remove it from the path string
-	if (fileExtension !== '') {
-		path = path.substring(0, path.length - fileExtension.length)
-	}
-	// Check if the path ends with ` (n)`
-	const suffix = path.match(/ \((\d+)\)$/) ? path.match(/ \((\d+)\)$/) : ''
-	// Initialise a pathwithout suffix variable
-	let pathWithoutSuffix = path
-	if (suffix !== '') {
-		// remove the suffix if any
-		pathWithoutSuffix = path.substring(0, path.length - suffix.length)
-	}
+	// If there is a file extension, remove it from the path string
+	path = path.replace(fileExtension, '')
```

Comment:
This will replace the file extension anywhere in the path.

So:
`Suggestion for new welcome.txt/welcome.txt` will turn into `Suggestion for new welcome/welcome (2).txt` note the missing `.txt` on the folder name

Reply:
My bad

Reply:
the regex does, but your replace does not

Reply:
Will work on it in different PR